afs/afs-local/src/main/java/com/powsybl/afs/local/storage/LocalCase.java

```diff
@@ -48,42 +55,56 @@ public String getPseudoClass() {
     }
 
     @Override
-    public String getStringAttribute(String name) {
-        switch (name) {
-            case "format":
-                return importer.getFormat();
-
-            case "description":
-                return importer.getComment();
-
-            default:
-                throw new AssertionError(name);
-        }
+    public String getDescription() {
+        return importer.getComment();
     }
 
     @Override
-    public OptionalInt getIntAttribute(String name) {
-        throw new AssertionError();
+    public NodeGenericMetadata getGenericMetadata() {
+        return new NodeGenericMetadata().setString("format", importer.getFormat());
     }
 
     @Override
-    public OptionalDouble getDoubleAttribute(String name) {
-        throw new AssertionError();
-    }
-
-    @Override
-    public Optional<Boolean> getBooleanAttribute(String name) {
-        throw new AssertionError();
+    public InputStream readBinaryData(String name) {
+        DataSource dataSource = Importers.createDataSource(file);
+        AppStorageDataSource.Name dataSrcName = AppStorageDataSource.Name.parse(name);
+        if (dataSrcName instanceof AppStorageDataSource.SuffixAndExtension) {
```

Comment:
You can reduce the complexity of this method if you put the try/catch outside the if/else blocks

Reply:
Done